### PR TITLE
Add typing for tmt/steps/provision/local.py

### PR DESCRIPTION
**Pull request by janhavlin**

### CODE REVIEW

File: tmt/steps/provision/local.py [hunk 1]
```diff
@@ -76,25 +80,32 @@ def execute(self, command, **kwargs):
...
-    def stop(self):
+    def stop(self) -> None:
         """ Stop the guest """
 
         self.debug(f"Doing nothing to stop guest '{self.guest}'.")
 
-    def reboot(self, hard=False, command=None, timeout=None):
+    def reboot(self,
+               hard: bool = False,
+               command: Optional[Union[List[str], str]] = None,
```
Comment (thrix) [inline tmt/steps/provision/local.py:90]: hmm, I wonder why `command` has here different type that in https://github.com/teemtee/tmt/blob/1920b07682e55c4d58fcbca76011505432e9fd76/tmt/steps/provision/__init__.py#L559?

**Review by thrix — CHANGES_REQUESTED** [review]
@janhavlin PTAL

File: tmt/steps/provision/local.py (hunk 1 above)
Comment (janhavlin) [reply tmt/steps/provision/local.py:90]: Yep, should be an `Optional[str]`. Likely a leftover from the time when `command` could be both `list` or `str`.

File: tmt/steps/provision/local.py [hunk 2]
```diff
@@ -42,21 +44,23 @@ def go(self):
...
-    def guest(self):
+    def guest(self) -> Optional['GuestLocal']:
         """ Return the provisioned guest """
         return self._guest
 
-    def requires(self):
+    def requires(self) -> List[Any]:
         """ List of required packages needed for workdir sync """
-        return GuestLocal.requires()
+        # TODO: revisit this cast once super class is typed
```
[the remark this answers is not in the document]
Comment (janhavlin) [reply tmt/steps/provision/local.py:53]: Removed.

**Review by thrix — APPROVED** [review]
LGTM

**Pull request merged**